### PR TITLE
Fixes for 17.0 mig report qweb pdf watermark

### PR DESCRIPTION
**Pull request by dz0**
There were 2 non-apparent issues

### CODE REVIEW

File: report_qweb_pdf_watermark/models/report.py [hunk 1]
```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
...
+            stream.seek(0, os.SEEK_SET)
+            header_byte = stream.read(5)
+            if header_byte != b"%PDF-":
+                # the file is not PDF -- probably this is image (we'll check bellow)
+                pdf_watermark = None
+            else:
+                # if file is PDF -- show the error (no idea where it may come from)
+                raise e
+
+        if not pdf_watermark:
```
Comment (oerp-odoo) [inline report_qweb_pdf_watermark/models/report.py]:
```suggestion
        if pdf_watermark is None:
```

File: report_qweb_pdf_watermark/models/report.py [hunk 2]
```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
...
+            # workaround: handle case of unintentional pypdf `UnicodeDecodeError`
+            # because odoo enforces `strict` mode
+            # https://github.com/versada/odoo/blob/d83bd4f/odoo/tools/pdf.py#L27
+            # and `pypdf` validation may can raise the error when generating message
+            # https://github.com/py-pdf/pypdf/blob/2.x/PyPDF2/_reader.py#L1455
+            stream = BytesIO(watermark)
+            stream.seek(0, os.SEEK_SET)
+            header_byte = stream.read(5)
+            if header_byte != b"%PDF-":
+                # the file is not PDF -- probably this is image (we'll check bellow)
```
Comment (oerp-odoo) [inline report_qweb_pdf_watermark/models/report.py:129]: Is this module supposed to handle images as well? Cause from name I would assume it only handles pdfs:)

File: report_qweb_pdf_watermark/models/report.py [hunk 3]
```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
...
+                pdf_watermark = None
+            else:
+                # if file is PDF -- show the error (no idea where it may come from)
+                raise e
```
Comment (oerp-odoo) [inline report_qweb_pdf_watermark/models/report.py]: Should be enough to just re-raise like:
```suggestion
                raise
```

File: report_qweb_pdf_watermark/models/report.py (hunk 2 above)
Comment (dz0) [reply report_qweb_pdf_watermark/models/report.py:129]: yes, 
- [code](https://github.com/versada/reporting-engine/pull/7/files#diff-66dceb063022cd58404dbbbb4649ffec677cbc89be10c385ad5714231929a220R138-R147)
- [test](https://github.com/versada/reporting-engine/blob/17.0-mig-report_qweb_pdf_watermark/report_qweb_pdf_watermark/tests/test_report_qweb_pdf_watermark.py#L22)

File: report_qweb_pdf_watermark/models/report.py [hunk 4]
```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
...
 
         pdf = PdfFileWriter()
-        pdf_watermark = None
+
         try:
             pdf_watermark = PdfFileReader(BytesIO(watermark))
         except PdfReadError:
-            # let's see if we can convert this with pillow
+            pdf_watermark = None
+        except UnicodeDecodeError as e:
```
Comment (dz0) [inline report_qweb_pdf_watermark/models/report.py]:
```suggestion
        except UnicodeDecodeError:
```

**Review by oerp-odoo — APPROVED** [review]

**Pull request merged**